### PR TITLE
Introduce Rendering Contexts

### PR DESCRIPTION
**Pull request by raphiz**
`PercyDOM.serialize` does not inline images and CSS that is referenced, eg `/assets/styles.123abc.css`.
When rendering the page in the docker image, relative assets (specifically images and CSS) cannot be loaded and the resulting screenshot isn't what we expect.

Therefore, this PR introduces a new concept of a "context" that contains the path to all those references.
It has certain drawbacks - especially regarding performance - but the main advantage is that it should work out of the box for most projects.

### CODE REVIEW

File: pixelpact/src/api.js [hunk 1]
```diff
@@ -46,8 +48,9 @@ export function buildFastify(renderFn, compareFn) {
       let actualHtml = request.body.actualHtml;
       let expected = Buffer.from(request.body.expected, "base64");
       let viewport = request.body.viewport;
+      let url = request.body.url ? request.body.url : "/";
```
Comment (mloydd) [inline pixelpact/src/api.js:57]:
```suggestion
      const` url = request.body.url ?? '/';
```

File: pixelpact/src/api.js [hunk 2]
```diff
@@ -68,8 +71,9 @@ export function buildFastify(renderFn, compareFn) {
     handler: async (request) => {
       let actualHtml = request.body.actualHtml;
       let viewport = request.body.viewport;
+      let url = request.body.url ? request.body.url : "/";
```
Comment (mloydd) [inline pixelpact/src/api.js:84]:
```suggestion
      const url = request.body.url ?? '/';
```

File: pixelpact/src/render.test.js [hunk 3]
```diff
@@ -50,32 +50,14 @@ describe("ContentServer", () => {
     contentServer.close();
   });
 
-  it("start prepares a temporary directory with an index.html", async () => {
-    await contentServer.start(anHtmlString);
+  it("serves the the given html contet at the given url after calling start", async () => {
```
Comment (mloydd) [inline pixelpact/src/render.test.js:53]:
```suggestion
  it("serves the given html content at the given url after calling start", async () => {
```

File: pixelpact/src/api.js [hunk 4]
```diff
@@ -45,12 +49,15 @@ export function buildFastify(renderFn, compareFn) {
       },
     },
     handler: async (request) => {
-      let actualHtml = request.body.actualHtml;
       let expected = Buffer.from(request.body.expected, "base64");
+      let actualHtml = request.body.actualHtml;
       let viewport = request.body.viewport;
       let url = request.body.url ? request.body.url : "/";
+      let context = request.body.context
```
Comment (mloydd) [inline pixelpact/src/api.js:59]:
```suggestion
      const context = request.body.context
```

**Review by mloydd — APPROVED** [review]

File: pixelpact/src/api.js (hunk 1 above)
Comment (raphiz) [reply pixelpact/src/api.js:57]: ha, did not know this nullish coalescing operator. Nice!

**Pull request merged**